stonesoup/feeder/multi.py

```diff
@@ -19,3 +23,133 @@ class MultiDataFeeder(DetectionFeeder, GroundTruthFeeder):
     @BufferedGenerator.generator_method
     def data_gen(self):
         yield from heapq.merge(*self.readers)
+
+
+class _QueueMultiDataFeeder(DetectionFeeder, GroundTruthFeeder):
+    reader = None
+    readers: Collection[Reader] = Property(doc='Readers to yield from')
+    max_size: int = Property(
+        default=0,
+        doc="Max queue size, where it will block more data being added. Default 0, unbounded.")
+
+    @cached_property
+    @abstractmethod
+    def _queue(self):
+        raise NotImplementedError
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._threads = None
+
+    @staticmethod
+    def _run(queue: Queue, reader: Reader):
+        for time, data in reader:
+            queue.put((time, data))
+
+    @BufferedGenerator.generator_method
+    def data_gen(self):
+        if self._threads is None:
+            self._threads = [
+                Thread(target=self._run, args=(self._queue, reader), daemon=True)
+                for reader in self.readers
+            ]
+            for thread in self._threads:
+                thread.start()
+
+        while any(thread.is_alive() for thread in self._threads) \
+                or not self._queue.empty():
+            try:
+                time, data = self._queue.get_nowait()
+            except Empty:
+                continue
+            yield time, data
+            self._queue.task_done()
+
+
+class FIFOMultiDataFeeder(_QueueMultiDataFeeder):
+    """FIFO Multi-data Feeder
+
+    This returns data from multiple data readers as a single stream,
+    where each reader is consumed in separate thread and put into a
```

Comment:
```suggestion
    where each reader is consumed in a separate thread and put into a
```
Not sure if this should be `a separate thread` or `separate threads`. The same applies to all class docstrings.